sources/packages/backend/apps/workers/src/controllers/disbursement/_tests_/e2e/disbursement.controller.associateMSFAA.e2e-spec.ts

```diff
@@ -0,0 +1,556 @@
+import {
+  createE2EDataSources,
+  createFakeMSFAANumber,
+  E2EDataSources,
+  MSFAAStates,
+  saveFakeApplication,
+  saveFakeApplicationDisbursements,
+  saveFakeSFASIndividual,
+  saveFakeStudent,
+} from "@sims/test-utils";
+import { DisbursementController } from "../../disbursement.controller";
+import { createTestingAppModule } from "../../../../../test/helpers";
+import { createFakeSFASPartTimeApplication } from "@sims/test-utils/factories/sfas-part-time-application";
+import { addDays, getISODateOnlyString } from "@sims/utilities";
+import { createFakeAssociateMSFAAPayload } from "./associate-MSFAA-payloads";
+import {
+  ApplicationStatus,
+  DisbursementScheduleStatus,
+  OfferingIntensity,
+} from "@sims/sims-db";
+import {
+  FakeWorkerJobResult,
+  MockedZeebeJobResult,
+} from "../../../../../test/utils/worker-job-mock";
+
+describe("DisbursementController(e2e)-associateMSFAA", () => {
+  let db: E2EDataSources;
+  let disbursementController: DisbursementController;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    db = createE2EDataSources(dataSource);
+    disbursementController = nestApplication.get(DisbursementController);
+  });
+
+  const now = new Date();
+
+  it("Should reuse valid signed MSFAA Number for part applications if its available for a student in SIMS when MSFAA is found for a student with part-time offering intensity.", async () => {
+    // Arrange
+    // Create student and save fake application with disbursements.
+    const student = await saveFakeStudent(db.dataSource);
+    const application = await saveFakeApplicationDisbursements(
+      db.dataSource,
+      { student },
+      {
+        offeringIntensity: OfferingIntensity.partTime,
+        applicationStatus: ApplicationStatus.Completed,
+      },
+    );
+
+    // Create and save fake MSFAA Number.
+    const savedMSFAANumber = await db.msfaaNumber.save(
+      createFakeMSFAANumber(
+        {
+          student: student,
+        },
+        {
+          msfaaState: MSFAAStates.Signed,
+
+          msfaaInitialValues: {
+            offeringIntensity: OfferingIntensity.partTime,
+          },
+        },
+      ),
+    );
+
+    // Act
+    const associateMSFAAPayload = createFakeAssociateMSFAAPayload({
+      assessmentId: application.currentAssessment.id,
+    });
+    const saveResult = await disbursementController.associateMSFAA(
+      associateMSFAAPayload,
+    );
+
+    // Asserts
+    expect(FakeWorkerJobResult.getResultType(saveResult)).toBe(
+      MockedZeebeJobResult.Complete,
+    );
+
+    // Fetch MSFAA Number for the student in SIMS.
+    const assignedMSFAANumber = await db.msfaaNumber.findOne({
```

Comment:
I am not following how the validation is being asserted here.
Should it not be retrieving the MSFAA using the MSFAA ID saved (associated) to the student application?